test/test_broker.hpp

```diff
@@ -1548,8 +1634,38 @@ class test_broker {
         // insert resend_messages here. get by for_each_store
         // PUBLISH with assigned packet_id
         // PUBREL
-        std::deque<offline_message> offline_messages;
+        std::list<offline_message> offline_messages;
```

Comment:
Please use multi_index that has `sequenced` and timer keys.
It is similar to 
https://github.com/redboltz/mqtt_cpp/blob/f9edd90ed064cc87f25ed36e7bd2013429ecad11/test/test_broker.hpp#L1035